packages/linear-progress/index.js

```diff
@@ -0,0 +1,160 @@
+import {MDCLinearProgressFoundation} from '@material/linear-progress/dist/mdc.linearProgress';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+class LinearProgress extends React.Component {
+  constructor(props) {
+    super(props);
+    this.bufferElement_ = null;
+    this.foundation_ = new MDCLinearProgressFoundation(this.adapter);
```

Comment:
This is done here in the `constructor` instead of in `componentDidMount` as it is then possible to unit test the `componentDidMount` implementation.

If this happens inside `componentDidMount` (as in many other instances) the instantiation overwrites the `testdouble` mock.

I understand that unit tests should never influence the design of a class/component/module, I felt in this case it didn't make much difference, while gaining the confidence of being able to unit test this piece.

Let me know if this is frowned upon and I will gladly change it, or maybe there is another way I am missing.

Reply:
No you're right...we lack unit tests for componentDidMount. We've also been putting the foundation on state. The team hasn't had the time to do analysis on whether or not it should go on state, moved the constructor, or in the DidMount. I've been moving some to state, since rendering child components is dependent on foundation.

To be more consistent with what we've been doing, try putting it on state instead. That way we only have 2 patterns instead of 3.

Reply:
I think what you can do, is {disableLifeCycleMethods: true}, then call mount() in the tests. Does that work? Otherwise keep it in the constructor for now to unblock you. Then we can hash out the details when you're ready for a review.

Reply:
Unfortunately not. I did try that at first. The issue with that approach is that enzyme sees a `shallow()` mounted component as _mounted_, and a component that is considered _mounted_ cannot be re-mounted.

From [the docs](https://airbnb.io/enzyme/docs/api/ReactWrapper/mount.html):
> A method that re-mounts the component, if it is not currently mounted.